### ISSUE TITLE
Simplify rounding utilities

### ISSUE THREAD

**Issue by LeaVerou** [shoelace-style/webawesome#253]
We currently have [utility classes for corner rounding](https://webawesome-git-next-font-awesome.vercel.app/docs/style-utilities/border-radius) that correspond to each of our `--wa-border-radius-*` design tokens. However, a design system is all about **restricting design choices** — we don't want people applying these at will. It seems that every element corresponds to a specific rounding (XS - M) with pill rounding being an additional option for some of the smaller components.

For non-WA elements, we can also decide what the optimal rounding can be based on their intended usage. We could even be smart and use `:has()` to ensure we always use a larger radius than any rounded descendants.

I’m not sure if we need `circle` either: `pill` has exactly the same affect on elements with an aspect ratio of 1, and on elements with a different aspect ratio it produces an ellipsis, which I'm not sure we want (also `circle` implies that it controls the aspect ratio too). That said, applying `pill` to get a circle feels confusing, but that could just indicate that we need a different name.

Therefore, it seems to me that all we need is two utility classes for rounding: `rounded` and `pill`. 

Going further, another realization is that if a theme has rounding, it seems that *all* elements with a visible shape need to be rounded, with the exact rounding depending on their specifics, contents etc. Crazy idea: What if Applied CSS would just apply that rounding preemptively, and if the element doesn't have any styling that makes its shape visible, no harm done. Then `rounded` would only be needed as an override. 

That said, we may still need the low-level rounding classes for edge cases and for custom elements, where we don't know what they are supposed to do. I would suggest we name them `rounding-*` rather than `border-radius-*` which is both longer, and [the name "border-radius" is considered one of the mistakes in CSS](https://wiki.csswg.org/ideas/mistakes). On that note, it wouldn't hurt to name our rounding tokens accordingly as well.

So to sum up, a strawman:
- `rounded` (or perhaps `rounding` to match the others?): High level, applies the appropriate rounding automatically
- Low-level, for edge cases: `rounding-0`, `rounding-xs`, `rounding-s`, `rounding-m`, `rounding-l`
- Something to replace `pill` so that it also works for circle. `rounding-max`? `rounding-circular`? `rounding-arc`? `round`? 🤔  Alternatively, we could just have two utilities that correspond to the same radius 🤷🏽‍♀️

Related to: #252

**Comment by lindsaym-fa** [comment]
Also related to #229.

`rounded`, `rounding-[0|xs|s|m|l]`, and `rounding-max` _plus_ `pill` (for folks inclined to reach for that term) make sense to me. I wonder if we'd want to cap the automatic rounding for `rounded` at `--wa-border-radius-l` so that outer containers don't get excessively bubbly?

Automatic rounding in applied.css is intriguing...

**Comment by LeaVerou** [comment]
> I wonder if we'd want to cap the automatic rounding for `rounded` at `--wa-border-radius-l` so that outer containers don't get excessively bubbly?

The way I envisioned it, `rounded` *always* corresponds to a certain rounding token (XS - L), it just automatically picks the right one. So yes, it would be capped anyway, since L is the largest radius (other than max).

**Comment by lindsaym-fa** [comment]
As discussed in [this Coda doc](https://coda.io/d/Design-Systems_dbS80bNLUcL/Patterns-observed-WIP_su3tfXIh#_luZ2Djho), we'll also remove the `xs` rounding property, shifting the scale to use `s` for badges etc, `m` for buttons etc, and `l` for cards etc.
